### PR TITLE
build: Update service to Keptn 0.12

### LINKED ISSUE #51: Update to Keptn 0.12
Upgrade go-utils, kubernetes-utils and the keptn/distributor to be compabitle with Keptn 0.12.

# Definition of done

- [ ] all Keptn related dependencies upgraded to 0.12.x
- [ ] Integration tested with Keptn 0.12.x

### PR DESCRIPTION
**Pull request by TannerGabriel**
Update the Argo-Service to be compatible with Keptn version `0.12.x`.

## Fixes
#51

### CODE REVIEW

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -52,13 +52,13 @@ spec:
           livenessProbe:
             httpGet:
               path: /health
-              port: 10999
+              port: 8080
```
Comment (pchila) [inline chart/templates/deployment.yaml:55]: On port 8080 is not the distributor responding but the argo service itself... I am not sure it's correct to use that as a distributor liveness/readiness probe

Comment (pchila) [reply chart/templates/deployment.yaml:55]: I saw that `prometheus-service` does the same and at the time I didn't think it was a big deal, now though I was thinking that if for whatever reason the service has a problem we are tearing down also the distributor and that doesn't seem right to me. As a counter-example in the `job-executor-service` the probes have been removed when upgrading to keptn 0.12 (see keptn-contrib/job-executor-service#173). 
@christian-kreuzberger-dtx  any thoughts/preferences/guidelines here ?

Comment (christian-kreuzberger-dtx) [reply chart/templates/deployment.yaml:55]: This is the best practice as defined by Keptn core:
distributor uses health check of port 8080, which is usually the service itself.

I agree that looks weird. i reached out to the Keptn team back then, and they said it's okay.

Comment (TannerGabriel) [reply chart/templates/deployment.yaml:55]: @christian-kreuzberger-dtx Thank you for letting us know. Then I will leave the health checks as they are now.

**Review by pchila — APPROVED** [review]
Given that the liveness/readinees probe setup is the one recommended by keptn core, the rest of the PR is ok :)

**Pull request merged**